index.js

```diff
@@ -174,16 +177,43 @@ Object.defineProperty(File.prototype, 'contents', {
   },
 });
 
+Object.defineProperty(File.prototype, 'cwd', {
+  get: function() {
+    return this._cwd;
+  },
+  set: function(cwd) {
+    if (typeof cwd !== 'string') {
+      throw new Error('cwd should be a string.');
+    }
+    this._cwd = endWithSep(normalize(cwd));
+  },
+});
+
+Object.defineProperty(File.prototype, 'base', {
+  get: function() {
+    return this._base || this._cwd;
+  },
+  set: function(base) {
+    if (base && typeof base !== 'string') {
+      throw new Error('base should be a string.');
```

Comment:
Couple more thoughts about this:

* I am fine with supporting `null` and `undefined` as values for `base` to "reset" to `cwd` resolution.
* I'm not sure I want to support an empty string as a "reset" value.  Empty string feels more like the user is saying "there is no base" for this file.  That feels a little weird (`base` should at least be the root, I think) so maybe it should throw on that.
* Other falsey values should not be allowed because this is a string method.  Throw on `0` or `false`.  If we throw on empty string, the check could just be `base == null || (typeof base === 'string' && base)`